### PR TITLE
Fix bullet titles and add anchors

### PR DESCRIPTION
**Pull request by harding**
This PR makes the newsletter content pass the tests described [here](https://github.com/bitcoinops/bitcoinops.github.io/pull/232#discussion_r332799560) and then adds those tests to the makefile.  By putting those bullet titles all on one line, they're automatically turned into anchors by a plugin added in a previous PR.

In a final commit, several anchors are manually added.  Content in the not-yet-added topics branches refers to those anchors.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,6 +21,10 @@ test-before-build:
 	! git --no-pager grep -L "^slug: " _posts
 	## Check that all slugs are unique
 	! git --no-pager grep -h "^slug: " _posts | sort | uniq -d | grep .
+	## Check for things that should probably all be on one line
+	@ ## Note: double $$ in a makefile produces a single literal $
+	! git --no-pager grep -- '^ *- \*\*[^\*]*$$'
```
Comment (jnewbery) [inline Makefile]: I don't think the backslash is necessary in the square brackets:

```
→ echo '*' | grep [*]; echo $?
*
0
→ echo '*' | grep [^*]; echo $?
1
```

I've verified that removing the `\` yields the same result when run against master.

File: _posts/en/newsletters/2018-12-28-newsletter.md [hunk 2]
```diff
@@ -325,6 +331,7 @@ recommended) using Tor---which can provide other benefits---but enabling
 encryption by default could help protect a larger number of users from
 eavesdropping by their ISPs.
 
+{:#untrackable-auth}
```
Comment (jnewbery) [inline _posts/en/newsletters/2018-12-28-newsletter.md]: Any reason not to use 'countersign' as the anchor here?

File: _posts/en/newsletters/2018-10-30-newsletter.md [hunk 3]
```diff
@@ -205,6 +205,7 @@ it to write a good description, and I doubt non-LN devs care -->{% endcomment %}
 - [LND #1981][] ensures that LND doesn't leak information about any of
   its peers that aren't advertising themselves as public nodes.
 
+{:#lnd-1535-1512}
```
Comment (jnewbery) [inline _posts/en/newsletters/2018-10-30-newsletter.md]: This splits the next bullet into its own list. Would it be better to just make `[LND #1535]` the id for the bullet?

File: _posts/en/newsletters/2019-06-12-newsletter.md [hunk 4]
```diff
@@ -38,10 +38,10 @@ changes to popular Bitcoin infrastructure projects.
       addresses, tracking when they've been paid, and finding or
       deriving the particular private keys necessary for spending.
 
-    - June 6th: discussion of the [consensus cleanup soft fork][xs
+    - <span id="cleanup-discussion" markdown="1">June 6th: discussion of the [consensus cleanup soft fork][xs
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-06-12-newsletter.md]: How about:

```
    - 
      {:#cleanup-discussion}
      June 6th: discussion of the [consensus cleanup soft fork][xs
```

Same for `assume-utxo-demo` below.

Comment (bitschmidty) [reply _posts/en/newsletters/2019-06-12-newsletter.md]: FWIW I couldn't get what John outlined to both work and pass the "things that should be on one line" check at the same.

Comment (harding) [reply _posts/en/newsletters/2019-06-12-newsletter.md]: @bitschmidty yeah, I had the same problems when trying different variations.  However, FYI, Block Attribute Lists (BALs) can go either before or after the block you want them to act on, so I was able to put it after the first paragraph of that bullet:

```text
    - June 6th: discussion of the [consensus cleanup soft fork][xs
      cleanup sf] included its interaction with [bip-taproot][], whether
      parts of it should be dropped, and whether anything should be
      added.  ([More background][bg consensus cleanup])
      {:#cleanup-discussion}
```

**Comment by harding** [comment]
Forced pushed edits for @jnewbery feedback (thanks!)

File: _posts/en/newsletters/2018-10-30-newsletter.md [hunk 5]
```diff
@@ -205,7 +205,8 @@ it to write a good description, and I doubt non-LN devs care -->{% endcomment %}
 - [LND #1981][] ensures that LND doesn't leak information about any of
   its peers that aren't advertising themselves as public nodes.
 
-- LND [#1535][LND #1535] and [#1512][LND #1512] adds the server-side
+- {:#lnd-1535-1512}
```
Comment (jnewbery) [inline _posts/en/newsletters/2018-10-30-newsletter.md]: This is still moving this one bullet point into its own `<ul>`. Can you move this Block Attribute List to after the text?

Comment (jnewbery) [reply _posts/en/newsletters/2018-10-30-newsletter.md]: actually, I've just done this myself and pushed

Comment (harding) [reply _posts/en/newsletters/2018-10-30-newsletter.md]: I think that edit is fine, but are you sure you were testing the updated branch?  On my branch (ede049d564559e65471b41c0027d7a1b20c939c9, without your change), I don't see an extraneous `<ul>`:

```html
  </li>
  <li id="lnd-1535-1512">
    <p>LND <a href="https://github.com/lightningnetwork/lnd/issues/1535">#1535</a> and <a href="https://github.com/lightningnetwork/lnd/issues/1512">#1512</a> adds the server-side
```

Comment (harding) [reply _posts/en/newsletters/2018-10-30-newsletter.md]: Yeah, looking at the GitHub forced push, it looks like you merged the old version of this with your one change.

**Pull request merged**